cloud-yarn/src/main/java/com/sequenceiq/cloudbreak/cloud/yarn/YarnMetadataCollector.java

```diff
@@ -138,7 +143,21 @@ private CloudResource getYarnApplcationResource(Iterable<CloudResource> resource
 
     @Override
     public List<CloudLoadBalancerMetadata> collectLoadBalancer(AuthenticatedContext ac, List<LoadBalancerType> loadBalancerTypes) {
-        // no-op
-        return Collections.emptyList();
+        YarnClient yarnClient = yarnClientUtil.createYarnClient(ac);
+        String loadBalancerApplicationName = applicationNameUtil.createLoadBalancerName(ac);
+        Iterable<Container> loadBalancerContainers = getContainers(loadBalancerApplicationName, yarnClient);
+        Iterator<Container> containerIterator = loadBalancerContainers.iterator();
+
+        List<CloudLoadBalancerMetadata> loadBalancerMetadata = Lists.newArrayList();
+        for (LoadBalancerType loadBalancerType : loadBalancerTypes) {
+            Container container = containerIterator.next();
```

Comment:
**question:** I don't think I'm following this method, will we always have the same number of Containers as `LoadBalancerType`s?

Here, the metadata is responsible for _assigning_ rather than looking up the container information?

Reply:
Good question, for Yarn this will always be true since Yarn will only ever have public loadbalancers.
In general though, as this method is for getting all of the loadbalancer metadata, and the only inpus is a list of the types, I kind of assume that at most there will be the same number of loadbalancers as the amount of unique types used. @hreeve-cloudera can clarify this some more.

I was just trying to make sure I future proof this in case Yarn does end up using private loadbalancers and the like in the future.

Reply:
It looks to me like in `LoadBalancerLaunchService`, you're creating one component per load balancer in the CloudStack, and each component has one container. So logically if there were two load balancers, there would be two components with a combined two containers. Is that right? That makes logical sense to me, and would match what seems to be being done in the for loop here.